deep-learning-server/examples/detector.c

```diff
@@ -0,0 +1,121 @@
+/*
+This file contains the functions for supporting detection (YOLO object 
+detection).
+
+AUTHORS
+
+The Veracruz Development Team.
+
+COPYRIGHT AND LICENSING
+
+See the `LICENSE_MIT.markdown` file in the Veracruz deep learning server 
+example repository root directory for copyright and licensing information.
+Based on darknet, YOLO LICENSE https://github.com/pjreddie/darknet/blob/master/LICENSE
+*/
+
+#include "darknet.h"
+
+
+// the function to test a object detector, aka detection inference. The prediction
+// is outputted as a `prediction` file
+// - Input: 1) data cfg (name of all objects), 2) network cfg, 3) weights file
+//          4) file to be tested (e.g., a image) 5) threshold for detection
+//          6) hierarchy threshold 7) output file path
+// - Output: None
+void test_detector(char *datacfg, char *cfgfile, char *weightfile, char *filename, float thresh, float hier_thresh, char *outfile)
+{
+    // read cfg file
+    list *options = read_data_cfg(datacfg);
+    char *name_list = option_find_str(options, "names", "data/names.list");
+    char **names = get_labels(name_list);
+
+    // load network
+    network *net = load_network(cfgfile, weightfile, 0);
+    set_batch_network(net, 1);
+
+    // load image
+    image **alphabet = load_alphabet();
+    double time;
+    char buff[256];
+    char *input = buff;
+    float nms=.45;
+
+    if(!filename){
+        fprintf(stderr, "file not exists: %s\n", filename);
+
+    }else{
+        // load image
+        strncpy(input, filename, 256);
```

Comment:
Why is this needed?  Can't you just pass `filename` to `load_image_color`?

Reply:
this `input` buff was considered for multiple images in streaming. For example, when a video is decoded as many images, the detector works for a bunch of images. I'm removing the buff, as it is not necessary for now.